ethereum/trie/src/main/java/org/hyperledger/besu/ethereum/trie/SimpleMerklePatriciaTrie.java

```diff
@@ -108,14 +110,17 @@ public void visitAll(final Consumer<Node<V>> nodeConsumer) {
   @Override
   public CompletableFuture<Void> visitAll(
       final Consumer<Node<V>> nodeConsumer, final ExecutorService executorService) {
-    nodeConsumer.accept(root);
     return CompletableFuture.allOf(
-        root.getChildren().stream()
-            .map(
-                rootChild ->
-                    CompletableFuture.runAsync(
-                        () -> rootChild.accept(new AllNodesVisitor<>(nodeConsumer)),
-                        executorService))
+        Stream.concat(
+                Stream.of(
+                    CompletableFuture.runAsync(() -> nodeConsumer.accept(root), executorService)),
```

Comment:
For this change, I think you need to bump up your queue size to 17 in the `Pruner`.  This is part of why I suggested moving this logic to the `Pruner` - it feels a bit tightly coupled in terms of how the traversal is happening and the required queue size.

Reply:
Maybe this is wrong but in my mental model, the 16 never mattered too much. It could very well be 2, or 7, or whatever. The reason being that the main thing we're expecting to do is to start marking sub-tries almost immediately through the `CallerRuns` policy.